### PR TITLE
Amber Sarayu volunteer tasks

### PR DESCRIPTION
**Pull request by AML441**
# Pull Request

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

### Code Review Preparation
- [ X] I have performed a self-review and reviewed the changed files
- [X ] My code follows the project's style guidelines and passes linting
- [ ] I have added comments in hard-to-understand areas and updated documentation

### Testing (Ignore for the first PR)
- [ ] I have added relevant tests and they pass locally
- [ ] I have tested the changes manually, including edge cases

### Code Quality & Security
- [ ] I have formatted my code and removed debugging/unused code
- [ X] No TypeScript errors or linting issues
- [ ] No sensitive information committed (API keys, passwords, etc.)
- [ ] User inputs are validated and sanitized

### Dependencies & Deployment
- [ ] Dependencies are documented and package-lock.json is updated if needed
- [ ] Changes work with Firebase emulator and environment variables are documentedRetry

## Screenshots (If it is a front end feature screenshot is required)

image
image
image

## Additional Notes

- I'm having issues with making the buttons on the very bottom of the page 
- Since the time block stuff hasn't been implemented, a lot of stuff that depends on the time block isn't there/hasn't been tested.

## Summary by CodeRabbit

* **New Features**
  * Added "Shift Tasks" for volunteers: a guided workflow to pick shifts, adjust inventory (add/remove), review selections, and confirm changes; supports dropping active shifts and validates inventory on confirm.
  * Volunteer-only navigation entry to access Shift Tasks.
  * Inline item search modal for selecting items and quantities.

* **Style**
  * New icons added and search icon refreshed; new slide-up animation for smoother UI transitions.

### CODE REVIEW

File: app/volunteer-tasks/page.tsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+                                {Object.keys(items).length > 0 && (
+                                    <button
+                                        onClick={() => setScreen("summary")}
+                                        className="mt-6 w-full bg-[#02AFC7] text-white"
+                                    >
+                                        Next
+                                    </button>
+                                )}
+                            </div>
```
[the remark this answers is not in the document]
Comment (AML441) [reply app/volunteer-tasks/page.tsx]: @coderabbitai review

[the remark this answers is not in the document]
Comment (AML441) [reply app/volunteer-tasks/page.tsx]: @coderabbitai review

File: components/volunteer-tasks/ItemSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+    onAdd({
+      name: selectedItem,
+      qty: Number(quantity),
+    });
+
+    setSelectedItem(null);
+    setQuantity("");
+    setSearchQuery("");
+  };
```
[the remark this answers is not in the document]
Comment (AML441) [reply components/volunteer-tasks/ItemSearch.tsx]: addressed in commit

File: app/volunteer-tasks/page.tsx [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
```
[the remark this answers is not in the document]
Comment (AML441) [reply app/volunteer-tasks/page.tsx]: i fiixed this with handleConfirm for summary page

File: app/volunteer-tasks/page.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (AML441) [reply app/volunteer-tasks/page.tsx]: the back button currently has no implementation since it's supposed to be timeblock screen

**Comment by AML441** [comment]
npm run build works fine in vscode but isn't working on github

**Review by joelchem — APPROVED** [review]
